src/js/button.js

```diff
@@ -7,13 +12,13 @@
  * @class
  * @constructor
  */
-vjs.Button = vjs.Component.extend({
+var Button = Component.extend({
   /**
    * @constructor
    * @inheritDoc
    */
   init: function(player, options){
```

Comment:
@carpasse theoretically, you shouldn't have any issues with the es6-changes beyond whatever breaking API changes we're having in 5.0. We definitely should have a doc and we might even make a small migration library.